pennylane_lightning/core/src/simulators/lightning_kokkos/gates/GateFunctorsParam.hpp

```diff
@@ -55,41 +60,43 @@ template <class Precision> struct multiQubitOpFunctor {
             wires_host(wires_.data(), wires_.size());
         Kokkos::resize(wires, wires_host.size());
         Kokkos::deep_copy(wires, wires_host);
-        dim = static_cast<std::size_t>(1U) << wires_.size();
+        dim = one << wires_.size();
         num_qubits = num_qubits_;
         arr = arr_;
         matrix = matrix_;
+        std::tie(parity, rev_wire_shifts) = wires2Parity(num_qubits_, wires_);
     }
 
     KOKKOS_INLINE_FUNCTION
     void operator()(const MemberType &teamMember) const {
-        const std::size_t k = teamMember.league_rank() * dim;
+        const std::size_t k = teamMember.league_rank();
         ScratchViewComplex coeffs_in(teamMember.team_scratch(0), dim);
         ScratchViewSizeT indices(teamMember.team_scratch(0), dim);
         if (teamMember.team_rank() == 0) {
-            Kokkos::parallel_for(
-                Kokkos::ThreadVectorRange(teamMember, dim),
-                [&](const std::size_t inner_idx) {
-                    std::size_t idx = k | inner_idx;
-                    const std::size_t n_wires = wires.size();
-
-                    for (std::size_t pos = 0; pos < n_wires; pos++) {
-                        std::size_t x =
-                            ((idx >> (n_wires - pos - 1)) ^
-                             (idx >> (num_qubits - wires(pos) - 1))) &
-                            1U;
-                        idx = idx ^ ((x << (n_wires - pos - 1)) |
-                                     (x << (num_qubits - wires(pos) - 1)));
-                    }
-
-                    indices(inner_idx) = idx;
-                    coeffs_in(inner_idx) = arr(idx);
-                });
+            std::size_t idx = (k & parity(0));
+            for (std::size_t i = 1; i < parity.size(); i++) {
+                idx |= ((k << i) & parity(i));
+            }
+            indices(0) = idx;
+            coeffs_in(0) = arr(idx);
```

Comment:
Is use of `(...)` faster than `[...]` here?

Reply:
I don't think there is a difference in performance, but parentheses are the official Kokkos syntax and I'm not sure brackets work for multidimensional indexing.